### PR TITLE
Create-basic-plugin

### PR DESCRIPTION
**Pull request by Pierregvx**

### CODE REVIEW

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+```
+
+:::info
+
+The `killTime` function logs a message every 5 seconds. It is used to show that the plugin is running.
+The `initializeAPI` function initializes the API and returns the server.
+
+:::
+
+- Still in `my-plugin.go`, set up the cors: the cors will allow the plugin to be called from the Massa Station server.
```
Comment (Thykof) [inline docs/tutorial/massa-station-plugin/create-basic-plugin.mdx:129]: first time you mentioned this file.

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+	<-intSig
+	quit <- true
+}
+```
+
+- Run the API on a local server to test and validate its functionality.
+
+```bash
+go run ./station-massa-hello-world.go
```
Comment (Thykof) [inline docs/tutorial/massa-station-plugin/create-basic-plugin.mdx:164]:
```
station-massa-hello-world.go:8:2: package my-plugin/station-massa-hello-world/api/server/restapi is not in GOROOT (/opt/homebrew/Cellar/go/1.20.6/libexec/src/my-plugin/station-massa-hello-world/api/server/restapi)
station-massa-hello-world.go:9:2: package my-plugin/station-massa-hello-world/api/server/restapi/operations is not in GOROOT (/opt/homebrew/Cellar/go/1.20.6/libexec/src/my-plugin/station-massa-hello-world/api/server/restapi/operations)
```

**Review by Thykof — CHANGES_REQUESTED** [review]

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx (hunk 2 above)
Comment (Pierregvx) [reply docs/tutorial/massa-station-plugin/create-basic-plugin.mdx]: should be good now

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx (hunk 1 above)
Comment (Pierregvx) [reply docs/tutorial/massa-station-plugin/create-basic-plugin.mdx:131]: fixed

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  this file will generate the API from the swagger file when you run `go generate ./... `
+
+  :::
+
+- Add the following code to it:
+
+  ```go
+  package main
+
+  //go:generate swagger generate server --quiet --target api/server --name station-massa-hello-world --spec api/swagger.yml --exclude-main
```
[the remark this answers is not in the document]
Comment (Thykof) [reply docs/tutorial/massa-station-plugin/create-basic-plugin.mdx]: ok good.

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+- Listen to the server created and register the plugin with the `RegisterPlugin` function from the plugin package.
+
+```go
+	listener, err := server.HTTPListener()
+	if err != nil {
+		panic(err)
+	}
+
+	plugin.RegisterPlugin(listener,plugin.Info{}) // registers the plugin
+```
```
Comment (Thykof) [inline docs/tutorial/massa-station-plugin/create-basic-plugin.mdx:196]: where should I put this code? in the main function?

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+:::info
+`plugin.RegisterPlugin` is a function available in the [plugin package](https://github.com/massalabs/station-massa-hello-world/blob/main/pkg/plugin/register.go).
+
+It facilitates the registration of a plugin with Massa Station, assigning a unique correlation ID to the plugin for identification purposes within the Massa Station system.
+
+:::
+
+### Install the plugin
+
+Build a binary of the plugin. `mkdir build && go build -o build/my-plugin`
```
Comment (Thykof) [inline docs/tutorial/massa-station-plugin/create-basic-plugin.mdx:207]: we need a proper go get before doing this 

```
my-plugin.go:15:2: no required module provides package github.com/massalabs/station-massa-hello-world/pkg/plugin; to add it:
	go get github.com/massalabs/station-massa-hello-world/pkg/plugin
```

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+Move the plugin binary, the manifest and the swagger file to the massastation plugin folder.
```
Comment (Thykof) [inline docs/tutorial/massa-station-plugin/create-basic-plugin.mdx:209]:
```suggestion
Move the plugin binary, the manifest and the swagger file to the Massa Station plugin folder.
```

**Review by Thykof — CHANGES_REQUESTED** [review]

File: docs/tutorial/massa-station-plugin/create-basic-plugin.mdx (hunk 4 above)
Comment (Pierregvx) [reply docs/tutorial/massa-station-plugin/create-basic-plugin.mdx:197]: yes, I added it

**Review by Thykof — APPROVED** [review]

**Pull request closed without merging**